### DISCUSSION TITLE
How to access user controller?

### DISCUSSION THREAD

**Discussion by adrian-ub** [strapi/strapi#7678] [Legacy Topics]
Hello friends,
I need to access the user controller and I have not been successful

what I need to do is from another controller call for example

```
await strapi.controllers.user.create(data.user);
```

**Comment by derrickmehaffy** [comment]
`await strapi.plugins['users-permissions'].controllers.user.create();` though I would advice against directly calling a controller if you aren't passing in the ctx from a request (ctx comes from koa)

**Reply by adrian-ub** [reply]
Thanks for answering, this way it has worked
But now I need to get the id, when I do it it returns an undefined

This is the code of what I try to do

```
async createStudentWithUser(ctx) {
    let ctx2 = ctx;
    ctx2.request.body = ctx.request.body.user;
    const user = await strapi.plugins[
      "users-permissions"
    ].controllers.user.create(ctx2);

    ctx.request.body.user = user.id;
    return await strapi.controllers.student.create(ctx);
  },
```

**Reply by derrickmehaffy** [reply]
I'm not sure I understand what you are trying to do? can you explain a bit more?

**Reply by adrian-ub** [reply]
What I am trying to do is create a "student" that has a relationship with "user"

So to avoid making two calls to the API, in this I send the information of both the student and the user, first register the user and then associate it with the student I need the id as the documentation says, since strapi does not supply this function

```
Student {
 user: User
 type: string
 ....blablabla other fields
}
```

in the POST of the api I am sending information similar to this

```
{
    user: {
      username: "asda",
      password: "1234567890",
      email: "asdadsasd@gmail.com",
      
    },
    universityProgram: "1",
    type: student
}
```

**Reply by adrian-ub** [reply]
I have solved it in the following way:
```
  async createStudentWithUser(ctx) {
    let ctx2 = ctx;

    ctx2.request.body = ctx.request.body.user;

    const user = await strapi.plugins[
      "users-permissions"
    ].controllers.user.create(ctx2);

    ctx.request.body.user = ctx2.response.body.id;

    return await strapi.controllers.student.create(ctx);
  },
```
But it returns null and not with the information of the created mode
